doc/ConfigureOptions.md

```diff
@@ -0,0 +1,123 @@
+  `key` is used. If the `key` is also not set, an error is thrown.
+* `default` a default value to use if the option is left blank. If no default
+  value is given the key in the config is left unset.
+
+Halibot further provides built-in subclasses of `Halibot.Option` for ease of
+use. They are as follows:
+
+* `Halibot.Option.String` for specifying an string option.
+* `Halibot.Option.Int` for specifying an integer option.
+* `Halibot.Option.Number` for specifying an real-valued number option.
+* `Halibot.Option.Bool` for specifying an boolean option.
+
+These option types are further made easily available via wrapper functions that
+simply wrap to the underlying `HalConfigurer.option`, and are respectively as
+follows:
+
+* `HalConfigurer.optionString(key=, prompt=, default=)`
+* `HalConfigurer.optionInt(key=, prompt=, default=)`
+* `HalConfigurer.optionNumber(key=, prompt=, default=)`
+* `HalConfigurer.optionBool(key=, prompt=, default=)`
+
+The `HalConfigurer` class also has a `validate` function, which consumes the
+user-defined configure function automatically and returns True or False
+depending on whether or not the currently specified config is valid. The
+validate function takes the optional parameter of `config` which will set the
+config to check, where otherwise it checks the config already associated with
+the configurer.
+
+Example module with Configurer
+------------------------------
+
+```python
+import time
+from halibot import HalModule, HalConfigurer
+
+class Responder(HalModule):
+
+	class Configurer(HalConfigurer):
+		def configure(self):
+			self.optionString(key='accept', prompt='Accept string')
+			self.optionString(key='return', prompt='Return string')
+			self.optionInt(key='delay', prompt='Wait seconds before responding', default=0)
+
+	def message(self, msg):
+		if 'accept' in self.config and msg.body == self.config['accept']:
+			time.sleep(self.config.get('delay', 0))
+			self.reply(msg, body=self.config['return'])
+```
+
+The Option class
+================
+
+The `Option` class allows you to write your own prompting or validation
+mechanisms for a specific object type.
+
+There are four methods one should define:
+
+* `configure(self, prompt)` which prompts the user for the value for this option
+  type with the given prompt. Sets a `self.value` to the result of the prompt by
+  default, calls `self.validate` to check, and reprompts if the validation failed.
+* `validate(self)` which returns true if the option's value is valid. Always
+  returns true by default.
+* `get(self)` which retrieves the JSON-compliant value set for this option.
+  Returns `self.value` by default.
+* `set(self, value)` which sets a value for this option from the JSON-compliant
+  value. Simply sets `self.value` by default.
+
+Example custom Option class
+---------------------------
+
+```python
+import halibot
+
+class OptAlphanumeric(halibot.Option.String):
+	def validate(self):
+		return self.value.isalpha()
+
```

Comment:
Maybe... it is certainly simpler, and I probably would prefer it for that, but I wonder if we lose anything. Such as, perhaps, the ability to translate a custom option type from an input string into some arbitrary representation of that data in an extensible manner ("true" => True is the example that comes to mind for me).